hxsl/Linker.hx

```diff
@@ -68,10 +69,31 @@ class Linker {
 		return Error.t(msg, p);
 	}
 
-	function mergeVar( path : String, v : TVar, v2 : TVar, p : Position ) {
+	function mergeVar( path : String, v : TVar, v2 : TVar, p : Position, shaderName : String ) {
 		switch( v.kind ) {
 		case Global, Input, Var, Local, Output:
 			// shared vars
+		case Param if ( shaderName != null ):
+			var valid = false;
+			for ( q in v.qualifiers ) {
+				switch (q) {
+					case Borrow(source):
+						if ( StringTools.startsWith(shaderName, source) )
+							valid = true;
+						break;
+					default:
+				}
+			}
+			for ( q in v2.qualifiers ) {
+				switch (q) {
+					case Borrow(source):
+						if ( StringTools.startsWith(shaderName, source) )
+							valid = true;
+						break;
+					default:
+				}
+			}
+			if (!valid) throw "assert";

```

Comment:
Why not use hasBorrowQualifer(v) || hasBorrowQualifier(v2) ?

Reply:
Mainly to avoid extra method calls. Can replace with method calls thought.

Reply:
Saving calls here is not much of an optimization ;)